server/src/test/java/io/crate/integrationtests/LogicalReplicationITest.java

```diff
@@ -500,4 +500,28 @@ public void test_alter_publication_add_table() {
             is("14768324| pub1| -450373579| false| doc| t1| true| true| true\n" +
                 "14768324| pub1| -450373579| false| doc| t2| true| true| true\n"));
     }
+
+    /**
+     * Ensures a dropped subscription can be immediately re-created.
+     * Specially, that all existing retention leases at the publisher shards are dropped when the subscription is dropped.
+     */
+    @Test
+    public void test_recreate_dropped_subscription() throws Exception {
```

Comment:
Not sure if it is really worth to have a itest for this. Creating a unit-test may be complex.